mlir/lib/Dialect/ArmSME/Transforms/TileAllocation.cpp

```diff
@@ -488,76 +501,126 @@ coalesceTileLiveRanges(DenseMap<Value, LiveRange> &initialLiveRanges) {
   return std::move(coalescedLiveRanges);
 }
 
-/// Choose a live range to spill (via some heuristics). This picks either an
-/// active live range from `activeRanges` or the new live range `newRange`.
+/// Choose a live range to spill (via some heuristics). This picks either a live
+/// range from `activeRanges`, `inactiveRanges`, or the new live range
+/// `newRange`. Note: All live ranges in `activeRanges` and `inactiveRanges` are
```

Comment:
`inactiveRanges` is a list of "holes"?

Reply:
No, it's a list of live ranges that are currently inactive (i.e. don't need to be in a register at this point in the program).

Reply:
I've added a comment explaining the active/inactive sets; 'holes' are the sections of a live range where it is inactive. 
We don't _directly_ track the holes (as in find all those spans), we just track what is active/inactive at the current point, which achieves the goal of filling the holes in the live ranges.